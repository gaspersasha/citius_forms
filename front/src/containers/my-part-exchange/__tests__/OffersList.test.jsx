import React from 'react';
import OffersListItem from '../OffersList';

describe('<OffersListItem />', () => {
  let wrapper;
  const props = {
    mileage: '12000',
    valuation: '2020',
    registration: 'WE33ADT',
    quoteId: 2,
  };

  it('rendered without errors', () => {
    wrapper = shallow(<OffersListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Pending if valuation equals null', () => {
    wrapper = mount(<OffersListItem {...props} valuation="" />);
    expect(wrapper.find('p[data-id="valuation"]').text()).toEqual('Pending');
  });
});
