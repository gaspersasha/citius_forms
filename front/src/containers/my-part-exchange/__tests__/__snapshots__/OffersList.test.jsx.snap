// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OffersListItem /> rendered without errors 1`] = `
<div
  className="pxOfferListItem"
>
  <div>
    <Typography
      align="left"
      className="paragraph"
      color="dark"
      type="p"
    >
      <b>
        Your Car
      </b>
    </Typography>
    <Typography
      align="left"
      className="paragraph"
      color="dark"
      type="p"
    >
      WE33ADT, 12000 miles
    </Typography>
  </div>
  <div>
    <Typography
      align="left"
      className="paragraph"
      color="dark"
      type="p"
    >
      <b>
        Your online offer
      </b>
    </Typography>
    <Typography
      align="left"
      className="paragraph"
      color="dark"
      data-id="valuation"
      type="p"
    >
      Â£2,020
    </Typography>
  </div>
  <Button
    className="button"
    disabled={false}
    href="/account/part-exchange/quote?2"
    onClick={null}
    selected={false}
    styleType="primary"
  >
    View offer
  </Button>
</div>
`;
